src/adapter/src/coord/cluster_scheduling.rs

```diff
@@ -0,0 +1,169 @@
+        &mut self,
+        decisions: Vec<(&'static str, Vec<(ClusterId, bool)>)>,
+    ) {
+        for (policy_name, decisions) in decisions.iter() {
+            for (cluster_id, decision) in decisions {
+                self.scheduling_decisions
+                    .entry(*cluster_id)
+                    .or_insert_with(Default::default)
+                    .insert(policy_name, *decision);
+            }
+        }
+
+        // Clean up those clusters from `scheduling_decisions` that
+        // - have been dropped, or
+        // - were switched to unmanaged, or
+        // - were switched to `SCHEDULE = MANUAL`.
+        for cluster_id in self.scheduling_decisions.keys().cloned().collect_vec() {
+            match self.get_managed_cluster_config(cluster_id) {
+                None => {
+                    // Cluster have been dropped or switched to unmanaged.
+                    self.scheduling_decisions.remove(&cluster_id);
+                }
+                Some(managed_config) => {
+                    if matches!(managed_config.schedule, ClusterScheduleOptionValue::Manual) {
+                        self.scheduling_decisions.remove(&cluster_id);
+                    }
+                }
+            }
+        }
+
+        for (cluster_id, decisions) in self.scheduling_decisions.clone() {
+            // If all policies have made a decision about this cluster
+            if POLICIES.iter().all(|policy| decisions.contains_key(policy)) {
+                // check whether the cluster's state matches the needed state.
+                let needs_replica = decisions.values().contains(&true);
+                let cluster_config = self
+                    .get_managed_cluster_config(cluster_id)
+                    .expect("cleaned up non-existing and unmanaged clusters above");
+                let has_replica = cluster_config.replication_factor > 0; // Is it on?
+                if needs_replica != has_replica {
+                    // Turn the cluster on or off.
+                    let mut new_config = cluster_config.clone();
+                    new_config.replication_factor = if needs_replica { 1 } else { 0 };
+                    if let Err(e) = self
+                        .sequence_alter_cluster_managed_to_managed(
+                            None,
+                            cluster_id,
+                            &cluster_config,
+                            new_config.clone(),
+                        )
+                        .await
+                    {
+                        soft_panic_or_log!(
+                            "handle_scheduling_decisions couldn't alter cluster {}. \
+                             Old config: {:?}, \
+                             New config: {:?}, \
+                             Error: {}",
+                            cluster_id,
+                            cluster_config,
+                            new_config,
+                            e
+                        );
```

Comment:
Not covered by tests according to coverage information. Can / shall we add a test for it?